src/seals/base_envs.py

```diff
@@ -122,33 +116,25 @@ def step(self, action: Action) -> Tuple[Observation, float, bool, dict]:
         return obs, rew, done, infos
 
 
-class ResettableMDP(ResettablePOMDP[State, State, Action], Generic[State, Action]):
-    """ABC for MDPs that are resettable."""
+class FullyObservableMixin(Generic[State]):
+    """Mixin for setting a POMDP as an MDP."""
 
-    def __init__(
-        self, *, state_space: gym.Space, action_space: gym.Space,
-    ):
-        """Build resettable MDP.
-
-        Args:
-            state_space: gym.Space containing possible states.
-            observation_space: gym.Space containing possible observations.
-                If None, defaults to `state_space`.
-            action_space: gym.Space containing possible actions.
-        """
-        super().__init__(
-            state_space=state_space,
-            observation_space=state_space,
-            action_space=action_space,
-        )
+    @property
+    def observation_space(self):
+        """Returns observation/state space."""
+        return self.state_space
 
     def obs_from_state(self, state: State) -> State:
-        """Identity since observation == state in an MDP."""
+        """Returns observation/state."""
         return state
 
 
-class TabularModelPOMDP(ResettableMDP[int, int]):
-    """Base class for tabular environments with known dynamics."""
+class TabularModelPOMDP(ResettablePOMDP[int, Observation, int], Generic[Observation]):
+    """Abstract class for tabular environments with known dynamics.
+
+    To make it concrete, you need to override at least observation_space and
```

Comment:
Can we provide an implementation of this, then? For example, using an observation matrix (which I think was in the original `TabularModelEnv` implementation)?

I think this would help validate this design. And if we don't implement this, then this PR hasn't bought us much in terms of actual features.

Reply:
I can add a test environment that implements a non-MDP environment, but I feel that TabularModelMDP is already an implementation that is testing TabularModelPOMDP?
Also, it is unclear to me what counts as a feature, but anyways, I consider this to be more of a refactor PR than a feature PR.

Reply:
What I'm getting at is that part of the stated motivation for this refactor is to allow tabular models to be partially observable. But we don't have any partially observable tabular model. It'd be easier to see if this API does/doesn't make that use case natural if we had a concrete example.

Reply:
I'm currently skeptical whether `FullyObservableMixin` is something you'd actually want to use as a mix-in. It implies you'd have a class with non-trivial implementation that is not fully observable, that you'd want to subclass and make fully observable. Seeing an example of it being used would help elucidate if this is actually an important use case or not.